src/main/java/org/terasology/metalrenegades/world/SimplexHillsAndMountainsProvider.java

```diff
@@ -62,20 +71,39 @@ public void process(GeneratingRegion region) {
         float[] mountainData = mountainNoise.noise(facet.getWorldArea());
         float[] hillData = hillNoise.noise(facet.getWorldArea());
         float[] mountainIntensityData = mountainIntensityNoise.noise(facet.getWorldArea());
+        float[] mesaData = mesaNoise.noise(facet.getWorldArea());
+        float[] mesaHeightData = mesaHeightNoise.noise(facet.getWorldArea());
 
         Biome[] biomeData = biomes.getInternal();
         float[] heightData = facet.getInternal();
         Iterator<Vector2ic> positions = facet.getWorldArea().iterator();
         for (int i = 0; i < heightData.length; ++i) {
+            // Noise values between threshold and threshold+smoothness are on the side of a mesa
+            // Noise values smaller than threshold are nothing, and larger than threshold+smoothness are on a steppe
+            float threshold = 0.5f;
+            float smoothness = 0.05f;
+            float mesaH = TeraMath.fadePerlin(TeraMath.clamp((mesaData[i] - threshold) / smoothness));
+            // Create a slope up to the bottom of the mesa
+            float slopeSteepness = 5;
+            float slopeHeight = 0.7f; // values from 0.5 - 1.5 look fine, might be noise later
+            float mesaL = TeraMath.clamp(slopeSteepness * (mesaData[i] - threshold - 0.5f * smoothness) + 1);
```

Comment:
The `- 0.5f * smoothness` is so that the slope ends exactly where the mesa wall starts (halfway past the threshold), so there's a smooth boundary instead of a jump.
Right, the `+ 1` is just for scaling, since we want the slope to appear on `x <- [threshold, 1]` instead of `x <- [threshold-1, 1]` which is what you'd get with just `x - threshold` (where `x` is the noise value).